src/main/scala/me/rexim/issuestant/github/EtagPolling.scala

```diff
@@ -1,29 +1,50 @@
 package me.rexim.issuestant.github
 
+import io.circe._
+import io.circe.generic.auto._
+import io.circe.parser._
+import io.circe.syntax._
+
 import org.http4s._
 import org.http4s.client._
+import org.http4s.circe._
 import org.http4s.util._
 
 import scalaz._
 import scalaz.stream._
 import scalaz.concurrent._
 
-class EtagPolling(client: Client, pollingUri: Uri) {
-  def responses: Process[Task, Response] =
-    Process.iterateEval(Response())(pollingIteration).tail.filter(_.status == Status.Ok)
+class EtagPolling[E](client: Client, pollingUri: Uri) {
+  // TODO(#65): make EtagPolling[E].Item a class, not a tuple
+  type Item = (Option[String], Option[E])
+
+  def responses(implicit decoder: Decoder[E]): Process[Task, E] =
+    Process
+      .iterateEval((None, None): Item)(pollingIteration(_)(decoder))
+      .collect { case (_, Some(e)) => e }
 
   private def getETag(response: Response): Option[String] =
     response.headers.get(CaseInsensitiveString("ETag")).map(_.value)
 
-  private def nextRequest(previousResponse: Response): Request =
+  private def nextRequest(previousItem: Item): Request =
     Request (
       uri = pollingUri,
       headers =
-        getETag(previousResponse)
+        previousItem._1
           .map(e => Headers(Header("If-None-Match", e)))
           .getOrElse(Headers.empty)
     )
 
-  private def pollingIteration(previousResponse: Response): Task[Response] =
-    client.fetch[Response](nextRequest(previousResponse))((x) => Task(x))
+  private def pollingIteration(previousItem: Item)(implicit decoder: Decoder[E]): Task[Item] =
+    client.fetch[Item](nextRequest(previousItem)) { (response) =>
+      implicit val http4sDecoder = jsonOf[E]
+
+      if (response.status == Status.Ok) {
+        response.as[E].map { e =>
+          (getETag(response), Some(e))
+        }
+      } else {
+        Task((getETag(response), None))
```

Comment:
We don't cover this case in UTs: https://coveralls.io/builds/12776098/source?filename=src%2Fmain%2Fscala%2Fme%2Frexim%2Fissuestant%2Fgithub%2FEtagPolling.scala#L47

This is why the coverage dropped a little bit. Covering that case may require significant changes, so creating a separate issue for that.